### ISSUE TITLE
Request for new features: 'Keep Forever Condition' and 'Action of Unkeep Forever, Then Delete After N Days'

### ISSUE THREAD

**Issue by nurgul212** [jenkinsci/build-history-manager-plugin#90]
### Feature Request
We propose adding the `Keep Forever Condition` and ` Action of Unkeep Forever (or Ignore Keep Forever)` options to the plugin's functionality. 
### Functionality
This new feature would allow users to: 

1. Apply a `Keep Forever` condition to certain builds
2. Implement an action of `Unkeep Forever or Ignore Keep Forever` after a specified number of days, then delete builds after a set period (N days) 
### Question to the Maintainer:

- Would you  be open to accepting and supporting this proposed feature if we were to develop and submit a pull request with the implementation?

**Comment by damianszczepanik** [comment]
`Keep Forever` is rather action, not condition. Or maybe `Keep build forever` so this is clear if the action refers to the build, artifact or log. Prepare PR for this, should be quite easy. Then check if https://github.com/jenkinsci/build-history-manager-plugin/blob/master/src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/DeleteBuildAction.java#L24 works as expected

**Comment by damianszczepanik** [comment]
Second looks like the opposite for the first one

**Comment by nurgul212** [comment]
It seems that there was a bit confusion in my previous comment. Let me clarify that we try to find a way to deal with runs that marked as `Keep Forever`. So we intend to propose an action like `Ignore Keep forever ` after a specified number of days, then delete builds after a set period (N days).

**Comment by nurgul212** [comment]
Right. We want to add a new feature called `Ignore or Uncheck Keep Forever`  as an action. This feature would automatically remove the `Keep Forever` tag from the builds that have it,  and then delete those builds after a specific number of days .

**Comment by damianszczepanik** [comment]
Then I would probably add new flag like 'forceDelete' to https://github.com/jenkinsci/build-history-manager-plugin/blob/master/src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/DeleteBuildAction.java instead of creating separate action. When the flag is selected then build toogle keep flag before deletion is invoked

**Comment by Khushigupta1830** [comment]
Hi, I am a beginner contributor preparing for GSoC 2026.
I would like to work on this issue.
Can someone please guide me where to start?
Thank you.

**Comment by damianszczepanik** [comment]
First we need to fully clarify what requester wants

**Comment by Khushigupta1830** [comment]
Yes , for that we need to connect.
How would you like to connect, for discussion?
